### PR TITLE
DAOS-15197 test: rebuild with 1 million files

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: RebuildMdtest
Skip-unit-tests: true
Skip-fault-injection-test: true

Add test to verify rebuild with 1 million files of size 1M.

Steps:
- Create pool / container
- Create 1 million EC_2P1GX files of size 1M
- Start reading the files
- Stop 1 rank during read
- Verify rebuild completes within some threshold
- Verify mdtest completes successfully

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by daltonbohning** [comment]
New test + affected passed with pmem:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13801/20/testReport/

Previously the new test passed multiple times with MD on SSD and pmem:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13801/19/testReport/FTEST_rebuild/RebuildMdtest/
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13801/18/testReport/FTEST_rebuild/RebuildMdtest/

**Review by phender — APPROVED** [review]

File: src/tests/ftest/rebuild/mdtest.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  control_method: daos
+
+mpirun:
+  args: "--bind-to hwthread --map-by socket"
+
+mdtest:
+  client_processes:
+    np: 64
+  api: DFS
+  dfs_oclass: EC_2P1GX
```
Comment (rpadma2) [inline src/tests/ftest/rebuild/mdtest.yaml:47]: I thought we use G1 type of oclass [eg: EC_2P1G1]... Is it not?

**Review by rpadma2 — APPROVED** [review]

File: src/tests/ftest/rebuild/mdtest.yaml (hunk 1 above)
Comment (daltonbohning) [reply src/tests/ftest/rebuild/mdtest.yaml:47]: It depends on what we're testing. GX will use all targets, so can be useful for rebuild testing. And in this case, the ticket specifically requested EC_2P1GX
https://daosio.atlassian.net/browse/DAOS-15197

File: src/tests/ftest/rebuild/mdtest.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      fabric_iface_port: 31417
+      log_file: daos_server1.log
+      log_mask: INFO
+      storage: auto
+
+pool:
+  size: 95%
+  properties: rd_fac:1
+  rebuild_timeout: 720  # 12 minutes
+  rebuild_check_inverval: 30  # check every 30 seconds
```
Comment (shimizukko) [inline src/tests/ftest/rebuild/mdtest.yaml:34]:
```suggestion
  rebuild_check_interval: 30  # check every 30 seconds
```
typo?

**Review by shimizukko — APPROVED** [review]

File: src/tests/ftest/rebuild/mdtest.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=pool,rebuild
+        :avocado: tags=RebuildMdtest,test_rebuild_mdtest
+        """
+        targets = self.server_managers[0].get_config_value("targets")
+        create_flags = self.params.get("create_flags", "/run/mdtest/*")
+        read_flags = self.params.get("read_flags", "/run/mdtest/*")
+        self.processes = self.params.get("np", "/run/mdtest/*", self.processes)
+        rebuild_check_inverval = self.params.get("rebuild_check_inverval", "/run/pool/*")
```
[the remark this answers is not in the document]
Comment (daltonbohning) [reply src/tests/ftest/rebuild/mdtest.py:35]: Oops :(
Thanks. Luckily it's only used in this one test. I'll fix in a followup if I don't need to push again

**Pull request merged**